hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -284,6 +285,7 @@ public class HoodieTableConfig extends HoodieConfig {
   public static final ConfigProperty<String> HIVE_STYLE_PARTITIONING_ENABLE = KeyGeneratorOptions.HIVE_STYLE_PARTITIONING_ENABLE;
 
   public static final List<ConfigProperty<String>> PERSISTED_CONFIG_LIST = Arrays.asList(
+      TIMESTAMP_TYPE_FIELD,
```

Comment:
Can you elaborate why this line is needed?

Reply:
https://github.com/apache/hudi/pull/4714/ started to persist all key generator props to table configuration, but missed `TIMESTAMP_TYPE_FIELD`.

It's strange that I could get even `hoodie.keygen.timebased.timestamp.scalar.time.unit` from `.hoodie.properties` if it's a `SCALAR` type of `TimestampKeyGenerator`, but couldn't get that `hoodie.keygen.timebased.timestamp.type`, which shows that it a `SCALAR` type. This change fix it.

I suppose the problem is, that there was a typo in initial MR https://github.com/apache/hudi/pull/4714/ in this part of the code:
```
  public static final List<String> PERSISTED_CONFIG_LIST = Arrays.asList(
      Config.DATE_TIME_PARSER_PROP,
      Config.INPUT_TIME_UNIT, Config.TIMESTAMP_INPUT_DATE_FORMAT_LIST_DELIMITER_REGEX_PROP,
      Config.TIMESTAMP_INPUT_DATE_FORMAT_PROP, Config.TIMESTAMP_INPUT_TIMEZONE_FORMAT_PROP,
      Config.TIMESTAMP_OUTPUT_DATE_FORMAT_PROP, Config.TIMESTAMP_OUTPUT_TIMEZONE_FORMAT_PROP,
      Config.TIMESTAMP_TIMEZONE_FORMAT_PROP, Config.DATE_TIME_PARSER_PROP
  );
```
Here `DATE_TIME_PARSER_PROP` is mentioned twice, but instead of the first `DATE_TIME_PARSER_PROP` should be `TIMESTAMP_TYPE_FIELD_PROP`, which is now named as `TIMESTAMP_TYPE_FIELD`.